### PR TITLE
Change the document's title when a page's visible changes

### PR DESCRIPTION
**Pull request by matthewp**
This is a start of #825, want to get feedback.

The approach is to use the `visible` property that is passed by
iron-pages to each page component, and when that changes to fire a
`document-title` event which is then handled by community-app.

Will update the other pages if this method is accepted.

### CODE REVIEW

File: client/src/catalog-community.html [hunk 1]
```diff
@@ -259,7 +260,7 @@ <h3 class="page-subtitle">Published on [[_formatDate(_content.metadata.original_
         if (!this._content || !this._content.metadata || !this._content.metadata.link)
```
Comment (samuelli) [inline client/src/catalog-community.html]: Let's also fire a title change here with the document title.

Some of the /community content is hosted, like the presentations. eg. https://www.webcomponents.org/community/presentations/progressive-performant-polymer-pick-three-at-google-io-2016 so we should update the title there too.

File: client/src/catalog-static.html [hunk 2]
```diff
@@ -80,6 +80,7 @@ <h1 class="page-title">[[pageTitle]]</h1>
 
         window.addEventListener('hashchange', this._hashChangeHandler);
         this.$.ajax.generateRequest();
+        this.fire('document-title', this.pageTitle);
```
Comment (samuelli) [inline client/src/catalog-static.html]: Maybe `if this.pageTitle`

**Comment by matthewp** [comment]
@samuelli Updated based on your feedback and changes a couple of other pages.

File: client/src/catalog-community.html [hunk 3]
```diff
@@ -256,10 +257,17 @@ <h3 class="page-subtitle">Published on [[_formatDate(_content.metadata.original_
       },
 
       _contentLoaded: function() {
-        if (!this._content || !this._content.metadata || !this._content.metadata.link)
+        if (!this._content || !this._content.metadata)
           return;
+        var metadata = this._content.metadata;
+        // there is no link, so set the title.
```
Comment (samuelli) [inline client/src/catalog-community.html]: Let's remove this comment, code is pretty readable

File: client/src/catalog-community.html [hunk 4]
```diff
@@ -256,10 +257,17 @@ <h3 class="page-subtitle">Published on [[_formatDate(_content.metadata.original_
...
+        if(!metadata.link) {
```
Comment (samuelli) [inline client/src/catalog-community.html]: Style nits: Here and elsewhere needs a space after `if`

**Comment by matthewp** [comment]
@samuelli Updated.

**Pull request merged**